### PR TITLE
Support rollback nodes in BlindingInfo

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingInfo.scala [hunk 1]
```diff
@@ -13,6 +13,9 @@ import com.daml.lf.value.Value.ContractId
   * "Disclosure" tells us which transaction nodes to communicate to which
   * parties.
   * See https://docs.daml.com/concepts/ledger-model/ledger-privacy.html
+  * Note that rollback nodes are a bit special here: Even if the node
+  * itself is not disclosed, a party will be disclosed if a node was below a
+  * rollback node. That information is not included in blinding info at this point.
```
Comment (cocreature) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingInfo.scala:18]: We might want to revisit this later but this is enough for the scenario service and sandbox classic and I’d rather start simple and add something later if we do end up needing it.

**Review by nickchapman-da — APPROVED** [review]
LGTM :+1: 

One small thing. I see that following the `GenActionNode` / `RollbackNode` split, we have removed `NodeInfo` from rollback nodes. So consequently, unlike other node types, there is no `informeesOfNode` (or `requiredAuthorizers`) for a rollback node. Is this correct?

**Comment by cocreature** [comment]
> One small thing. I see that following the `GenActionNode` / `RollbackNode` split, we have removed `NodeInfo` from rollback nodes. So consequently, unlike other node types, there is no `informeesOfNode` (or `requiredAuthorizers`) for a rollback node. Is this correct?

Exactly, only actions have informees. Rollback nodes are not actions. However, we can still disclose a rollback node to the informees of the parent exercise.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/ledger/BlindingTransaction.scala [hunk 2]
```diff
@@ -57,51 +57,55 @@ object BlindingTransaction {
...
+          // nodes of every type are disclosed to their witnesses
+          val state = state0.discloseNode(witnesses, nodeId)
 
-      action match {
+          action match {
 
-        case _: Node.NodeCreate[ContractId] => state
-        case _: Node.NodeLookupByKey[ContractId] => state
+            case _: Node.NodeCreate[ContractId] => state
+            case _: Node.NodeLookupByKey[ContractId] => state
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/ledger/BlindingTransaction.scala:71]:
```suggestion
    case _: Node.NodeCreate[ContractId] |  _: Node.NodeLookupByKey[ContractId] => state
```

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/ledger/BlindingTransaction.scala [hunk 3]
```diff
@@ -57,51 +57,55 @@ object BlindingTransaction {
...
+          // Rollback nodes are disclosed to the witnesses of the parent exercise.
+          val state = state0.discloseNode(parentExerciseWitnesses, nodeId)
+          rollback.children.foldLeft(state) { (s, childNodeId) =>
             processNode(
               s,
-              witnesses,
+              parentExerciseWitnesses,
               childNodeId,
             )
           }
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/ledger/BlindingTransaction.scala:103]:
``` suggestion
          rollback.children.foldLeft(state) ((s, childNodeId) =>
            processNode(
              s,
              parentExerciseWitnesses,
              childNodeId,
            )
          )
```

or 

``` suggestion
          rollback.children.foldLeft(state)(processNode(_, parentExerciseWitnesses, _))
```

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingInfo.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingInfo.scala:18]: I agree, but I would have added a  `// TODO https://github.com/digital-asset/daml/issues/8020` with this comment inside the code.

**Review by remyhaemmerle-da — COMMENTED** [review]
LGTM

**Comment by remyhaemmerle-da** [comment]
part of #8020

**Pull request merged**